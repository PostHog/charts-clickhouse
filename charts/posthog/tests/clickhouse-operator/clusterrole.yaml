suite: ClickHouse cluster role definition
templates:
  - templates/clickhouse_cluster_role.yaml

tests:
  - it: should be empty if clickhouseOperator.enabled is set to false
    set:
      clickhouseOperator.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: the manifest should match the snapshot when using default values
    asserts:
      - hasDocuments:
          count: 2
      - matchSnapshot: {}

  - it: ClusterRoleBinding/clickhouse-operator-posthog-cr subjects[0].namespace override via clickhouseOperator.namespace should work
    set:
      clickhouseOperator.namespace: "custom-namespace"
    documentIndex: 1
    asserts:
      - hasDocuments:
          count: 2
      - equal:
          path: subjects[0].namespace
          value: custom-namespace

  - it: ClusterRoleBinding/clickhouse-operator-posthog-cr subjects[0].namespace override via Release.Namespace should work
    release:
      namespace: custom-namespace-from-release
    documentIndex: 1
    asserts:
      - hasDocuments:
          count: 2
      - equal:
          path: subjects[0].namespace
          value: custom-namespace-from-release
