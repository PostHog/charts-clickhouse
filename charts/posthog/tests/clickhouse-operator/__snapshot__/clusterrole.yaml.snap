the manifest should match the snapshot when using default values:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        clickhouse.altinity.com/chop: 0.19.0
      name: clickhouse-operator-NAMESPACE
      namespace: NAMESPACE
    rules:
    - apiGroups:
      - ""
      resources:
      - configmaps
      - services
      verbs:
      - get
      - list
      - patch
      - update
      - watch
      - create
      - delete
    - apiGroups:
      - ""
      resources:
      - endpoints
      verbs:
      - get
      - list
      - watch
    - apiGroups:
      - ""
      resources:
      - events
      verbs:
      - create
    - apiGroups:
      - ""
      resources:
      - persistentvolumeclaims
      verbs:
      - get
      - list
      - patch
      - update
      - watch
      - delete
    - apiGroups:
      - ""
      resources:
      - persistentvolumes
      - pods
      verbs:
      - get
      - list
      - patch
      - update
      - watch
    - apiGroups:
      - apps
      resources:
      - statefulsets
      verbs:
      - get
      - list
      - patch
      - update
      - watch
      - create
      - delete
    - apiGroups:
      - apps
      resources:
      - replicasets
      verbs:
      - get
      - patch
      - update
      - delete
    - apiGroups:
      - apps
      resourceNames:
      - clickhouse-operator
      resources:
      - deployments
      verbs:
      - get
      - patch
      - update
      - delete
    - apiGroups:
      - policy
      resources:
      - poddisruptionbudgets
      verbs:
      - get
      - list
      - patch
      - update
      - watch
      - create
      - delete
    - apiGroups:
      - clickhouse.altinity.com
      resources:
      - clickhouseinstallations
      verbs:
      - get
      - patch
      - update
      - delete
    - apiGroups:
      - clickhouse.altinity.com
      resources:
      - clickhouseinstallations
      - clickhouseinstallationtemplates
      - clickhouseoperatorconfigurations
      verbs:
      - get
      - list
      - watch
    - apiGroups:
      - clickhouse.altinity.com
      resources:
      - clickhouseinstallations/finalizers
      - clickhouseinstallationtemplates/finalizers
      - clickhouseoperatorconfigurations/finalizers
      verbs:
      - update
    - apiGroups:
      - clickhouse.altinity.com
      resources:
      - clickhouseinstallations/status
      - clickhouseinstallationtemplates/status
      - clickhouseoperatorconfigurations/status
      verbs:
      - get
      - update
      - patch
      - create
      - delete
    - apiGroups:
      - ""
      resources:
      - secrets
      verbs:
      - get
      - list
    - apiGroups:
      - apiextensions.k8s.io
      resources:
      - customresourcedefinitions
      verbs:
      - get
      - list
